plugins/getplants.cpp

```diff
@@ -224,6 +231,106 @@ selectability selectablePlant(const df::plant_raw *plant, bool farming)
     }
 }
 
+//  Formula for determination of the variance in plant growth maturation time, determined via disassembly.
+//  The x and y parameters are in tiles relative to the embark.
+bool ripe(int32_t x, int32_t y, int32_t start, int32_t end) {
+    int32_t time = ((435522653 - (((y + 3) * x + 5) * ((y + 7) * y * 400181475 + 289700012))) & 0x3FFFFFFF) % 2000 + (*cur_year_tick % 403200);
```

Comment:
The `% 403200` should be applied _after_ adding `cur_year_tick` to the delta value (since `cur_year_tick` is already guaranteed to be within that range).

Reply:
Thanks. It looked odd.